src/ophyd_async/core/detector.py

```diff
@@ -171,7 +173,7 @@ async def trigger(self) -> None:
         indices_written = await self.writer.get_indices_written()
         written_status = await self.controller.arm(DetectorTrigger.internal, num=1)
         await written_status
-        await self.writer.wait_for_index(indices_written + 1)
+        await self.writer.wait_for_index(indices_written + 1, timeout=DEFAULT_TIMEOUT)
```

Comment:
It may be better to pass the timeout into this object rather than always use the default, because that would be consistent with the other changes in this PR

Reply:
So the writer.wait_for_timeout method gets called twice in the code base, once here and once for the flyer. In the flyer, I have a _trigger_to_frame_timeout on the Flyer that can be set, which gets propagated down to this method call. But here, I don't. I think I agree with you; I'll make the detector have a _detector_index_timeout or something similar to indicate this.

Reply:
I would call it `_frame_wrote_timeout`

Reply:
`_frame_writing_timeout` ?

Reply:
Sure